test/e2e/clusterctl_upgrade_test.go

```diff
@@ -283,8 +283,8 @@ var _ = Describe("When testing clusterctl upgrades (v1.6=>current)", func() {
 			InitWithBinary:            fmt.Sprintf(clusterctlDownloadURL, stableRelease),
 			InitWithProvidersContract: "v1beta1",
 			//  Note: Both InitWithKubernetesVersion and WorkloadKubernetesVersion should be the highest mgmt cluster version supported by the source Cluster API version.
-			InitWithKubernetesVersion: "v1.29.2",
-			WorkloadKubernetesVersion: "v1.29.2",
+			InitWithKubernetesVersion: "v1.29.4",
```

Comment:
(I guess it's fine for now because kind didn't really change, but I think in CAPI v1.6 we wouldn't pin to the right sha this way, so if the next kind version breaks something & publisheds v1.29.4 stuff breaks)

Reply:
reverted